src/ansys/simai/core/data/models.py

```diff
@@ -38,6 +114,43 @@ class ModelConfiguration:
     simulation_volume: Dict[str, Any] = None
     project_id: str = None
 
+    @property
+    def domain_of_analysis(self) -> DomainOfAnalysis | None:
+        """The Domain of Analysis of the model configuration."""
+        if not self.simulation_volume:
+            return None
+
+        doa = DomainOfAnalysis()
+
+        doa.Length = self._get_doa_axis("X")
+        doa.Width = self._get_doa_axis("Y")
+        doa.Height = self._get_doa_axis("Z")
+
+        return doa
+
+    @domain_of_analysis.setter
+    def domain_of_analysis(self, doa: DomainOfAnalysis):
+        """Sets the Domain of Analysis in the configuration."""
+
+        if self.simulation_volume is None:
+            self.simulation_volume = {}
+
+        self.simulation_volume["X"] = self._set_doa_axis(doa.Length, "Length")
+        self.simulation_volume["Y"] = self._set_doa_axis(doa.Width, "Width")
+        self.simulation_volume["Z"] = self._set_doa_axis(doa.Height, "Height")
+
+    def _get_doa_axis(self, rel_pos: str) -> DomainAxisDefinition:
+        pos = self.simulation_volume.get(rel_pos)
+        return DomainAxisDefinition(position=pos["type"], value=pos["value"], length=pos["length"])
+
+    def _set_doa_axis(self, fld: DomainAxisDefinition, param: str) -> Dict[str, Any]:
```

Comment:
Why not directly use the serialization/deserialization features of pydantic ?

if `fld` can be `None` then its type should be `Optional[fld]` (is `fld` shorthand for "field" ?)

Reply:
1. regarding serialization/deserialization with pydantic, I replace here `type` to position but I could indeed use `BaseModel.model_dump(by_alias=True)` in `_set_doa_axis` and a `serialization_alias` for `position`, I haven't thought of it. Nevertheless, I don't think I could use any pydantic feature that converts "type" to "position" in `_get_doa_axis`.

2. `fld` is not optional. I check, however, if it is `None` and raise an exception to avoid user errors when partially setting DoA, e.g., the `Length` is defined but the `Width` and the `Height` are not. Adding `Optional` here would be misleading.